tiingo/wsclient.py

```diff
@@ -0,0 +1,93 @@
+import os
+import websocket
+try:
+    import thread
+except ImportError:
+    import _thread as thread
+import time
+import json
+
+GLOB_config=None
+GLOB_on_msg_cb=None
+
+class genericWebsocketClient:
+    '''
+    the methods passed to websocketClient have to be unbounded if we want WebSocketApp to pass everything correctly
+    see websocket-client/#471
+    '''
+    def on_message(ws, message):
+        GLOB_on_msg_cb(message)
+    def on_error(ws, error):
+        print(error)
+    def on_close(ws):
+        pass
+    def on_open(ws):
+        def run(*args):
+            print(GLOB_config)
+            ws.send(json.dumps(GLOB_config))
+        thread.start_new_thread(run, ())
+    def __init__(self,config,on_msg_cb):
+        global GLOB_config
```

Comment:
Hey, thanks for investigating this! I'm a little bit concerned by the use of `global` here, so I went and looked at the issue in the comment, and I'm not sure that I entirely follow why resolving this requires using globals. https://github.com/websocket-client/websocket-client/issues/471

I was wondering if there were any existing Websocket clients that this implementation was inspired by that I could take a look at.

Reply:
i have tracked this down to this [line of code](https://github.com/websocket-client/websocket-client/blob/29c15714ac9f5272e1adefc9c99b83420b409f63/websocket/_app.py#L343) in the  library.
which checks if the method is bounded before passing ```self``` to it which is needed for the number of arguments to match up. and ofc you can't have unbounded method access class data. I would love not to use ```global``` ,but i have no idea how to implement something like that without actually changing the ```websocket-client``` code. if you have an idea or if i am missing something, Please. let me know

Reply:
I guess what I'm unsure about is why `genericWebsocketClient` class is necessary in the first place, since we never actually even read the value in the `ws_client` variable. This is why I was curious if this is based on an existing production library / program that you saw, so I can evaluate whether that approach makes sense for us too.

If we're not actually going to read the value of `ws_client`, I'm not sure that I see the value of creating it in the first place.